tasks/EchoesToFrameConverterTask.hpp

```diff
@@ -1,46 +1,63 @@
 /* Generated from orogen/lib/orogen/templates/tasks/Task.hpp */
 
-#ifndef RADAR_BASE_TASK_TASK_HPP
-#define RADAR_BASE_TASK_TASK_HPP
-
-#include "radar_base/TaskBase.hpp"
-
-namespace radar_base{
-
-    /*! \class Task
-     * \brief The task context provides and requires services. It uses an ExecutionEngine to perform its functions.
-     * Essential interfaces are operations, data flow ports and properties. These interfaces have been defined using the oroGen specification.
-     * In order to modify the interfaces you should (re)use oroGen and rely on the associated workflow.
-     * Declare a new task context (i.e., a component)
-
-The corresponding C++ class can be edited in tasks/Task.hpp and
-tasks/Task.cpp, and will be put in the radar_base namespace.
+#ifndef RADAR_BASE_ECHOESTOFRAMECONVERTERTASK_TASK_HPP
+#define RADAR_BASE_ECHOESTOFRAMECONVERTERTASK_TASK_HPP
+
+#include "radar_base/EchoToImageLUT.hpp"
+#include "radar_base/EchoesToFrameConverterTaskBase.hpp"
+#include <base/Time.hpp>
+#include <base/samples/Frame.hpp>
+#include <memory>
+#include <opencv2/highgui.hpp>
+
+namespace radar_base {
+
+    /*! \class EchoesToFrameConverterTask
+     * \brief The task context provides and requires services. It uses an ExecutionEngine
+     to perform its functions.
+     * Essential interfaces are operations, data flow ports and properties. These
+     interfaces have been defined using the oroGen specification.
+     * In order to modify the interfaces you should (re)use oroGen and rely on the
+     associated workflow.
+     *
      * \details
      * The name of a TaskContext is primarily defined via:
      \verbatim
      deployment 'deployment_name'
-         task('custom_task_name','radar_base::Task')
+         task('custom_task_name','radar_base::EchoesToFrameConverterTask')
      end
      \endverbatim
-     *  It can be dynamically adapted when the deployment is called with a prefix argument.
+     *  It can be dynamically adapted when the deployment is called with a prefix
+     argument.
      */
-    class Task : public TaskBase
-    {
-	friend class TaskBase;
-    protected:
+    class EchoesToFrameConverterTask : public EchoesToFrameConverterTaskBase {
+        friend class EchoesToFrameConverterTaskBase;
+
+    private:
+        int m_current_sweep_size = 0;
+        int m_current_num_angles = 0;
+        float m_current_range = 0;
+        base::Time m_last_sample;
 
+        std::map<int, std::unique_ptr<std::vector<uint8_t>>> m_echoes;
```

Comment:
What ? No ... no ... no ................. I know it works and probably performance is not too horrible given the sizes, but I can't let something like this pass.

First of all, even if map would be a good tool here, the unique_ptr makes no sense.

However, the discretized angles form a contiguous space of integers. The worst I thought I could encounter here would be a `std::vector<std::vector<uint8_t>>`

But even that is bad, as all vectors will have the same size.

Just manage it as a flat `std::vector<uint8_t>` initialized at zero, where a beam at angle i is in `[n * beam_size, (n + 1) * beam_size - 1]`